### PR TITLE
fix(fluid-build): Handle npm aliases in deps install check

### PR DESCRIPTION
**Pull request by tylerbutler**
fluid-build contains code to check if dependencies are installed. This is used in several places throughout build-tools, but it doesn't handle the aliases added in #21884.

I see errors like the following on MacOS in some cases - most notably when I make lockfile changes.

```
ERROR: @fluidframework/build-tools: dependency @fluidframework/build-tools-bin not found
ERROR: @fluidframework/bundle-size-tools: dependency @fluidframework/build-tools-bin not found
ERROR: @fluid-tools/version-tools: dependency @fluidframework/build-tools-bin not found
ERROR: Dependency not installed. Use --install to fix.
``` 

With this change, the code now falls back to checking the alias path if the primary path isn't found.

### CODE REVIEW

File: build-tools/packages/build-tools/src/common/npmPackage.ts [hunk 1]
```diff
@@ -275,11 +277,23 @@ export class Package {
...
 				!lookUpDirSync(this.directory, (currentDir) => {
-					// TODO: check semver as well
-					return existsSync(path.join(currentDir, "node_modules", dep.name));
+					if (existsSync(path.join(currentDir, "node_modules", dep.name))) {
+						return true;
+					}
+					const alias = dep.version.match(regexNpmAlias)?.groups?.alias;
+					return alias === undefined
+						? false
+						: existsSync(path.join(currentDir, "node_modules", alias));
```
Comment (jason-ha) [inline build-tools/packages/build-tools/src/common/npmPackage.ts:296]: The existing logic is correct. pnpm should be placing a directory using the "key" name. The alias aspect should only be impacting the content of that directory.

**Review by jason-ha — CHANGES_REQUESTED** [review]
Not requiring peers seems okay, but that is not the issue.
Not sure why Mac would have an issue. I would have thought the pnpm install structure should be the same across OSes. Suggest wiping build-tools and re-`pnpm i`ing. (I'd say use `--frozen-lockfile` but that should be the default and should not impact "hoisting" that we don't use.)

**Comment by tylerbutler** [comment]
> Not requiring peers seems okay, but that is not the issue.
> 
> Not sure why Mac would have an issue. I would have thought the pnpm install structure should be the same across OSes. Suggest wiping build-tools and re-`pnpm i`ing. (I'd say use `--frozen-lockfile` but that should be the default and should not impact "hoisting" that we don't use.)

I get a consistent repro even in a fresh clone. I tried a clone on a different volume with an empty Pnpm cache too, same result.

I only have one Mac so I haven't been able to try different systems. I'll ask others to try to repro on Monday. I have been having odd problems with hoisting on another project on Pnpm 9, and now I'm wondering if it's all related. I'll poke around the pnpm issues list too. Maybe there's something there.

File: build-tools/packages/build-tools/src/common/npmPackage.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/common/npmPackage.ts:296]: With fresh eyes this is clearly wrong; thanks. I'll keep looking into Pnpm.

**Pull request closed without merging**